src/main/java/com/amazon/opendistroforelasticsearch/security/user/AuthCredentials.java

```diff
@@ -175,19 +176,16 @@ public int hashCode() {
     public boolean equals(Object obj) {
         if (this == obj)
             return true;
-        if (obj == null)
-            return false;
-        if (getClass() != obj.getClass())
+        if (obj == null || getClass() != obj.getClass()) {
             return false;
+        }
+
         AuthCredentials other = (AuthCredentials) obj;
-        if (internalPasswordHash == null || other.internalPasswordHash == null || !MessageDigest.isEqual(internalPasswordHash, other.internalPasswordHash))
-            return false;
-        if (username == null) {
-            if (other.username != null)
-                return false;
-        } else if (!username.equals(other.username))
-            return false;
-        return true;
+        return Objects.equals(username, other.username)
+            && Objects.equals(nativeCredentials, other.nativeCredentials)
+            && Objects.equals(backendRoles, other.backendRoles)
+            && MessageDigest.isEqual(internalPasswordHash, other.internalPasswordHash)
+            && Objects.equals(attributes, other.attributes);
```

Comment:
Add `&& Objects.equals(password, other.password)` here?

Reply:
The password is always filled with nulls and then set to null in the constructor, and the internalPasswordHash is used instead. We can check it here though if you prefer.

Reply:
Is it possible to be non-null values in any scenarios? If so, I'd add a check for it.

Reply:
It is not possible, but I can still add the check, is that what you'd prefer?

Reply:
Yes, let's add the check. Thanks!